tachyon/math/base/semigroups.h

```diff
@@ -215,6 +215,22 @@ class MultiplicativeSemigroup {
     return ret;
   }
 
+  constexpr static std::pmr::vector<MulResult> GetSuccessivePowersPmr(
```

Comment:
What about keeping only one `GetSuccessivePowers()` that uses `std::pmr::vector`? If we have two versions, users might get confused. Additionally, this typically generates long series of powers, so having a single, efficient implementation would be beneficial.

Reply:
@chokobole  The scope is `math` so I thought there can be other use cases for this as well.

Reply:
I was thinking of cases when we use small size. But I guess we don't care about performance much in those cases. I will remove it in separate commit since it will be big change.